Multi modal RAG + QA Cookbooks

rlancemartin opened the pull request.

rlancemartin committed:
Multi modal RAG Cookbook

eyurtsev committed:
Fix for anyscale chat model api key (#12938)

* ChatAnyscale was missing coercion to SecretStr for anyscale api key
* The model inherits from ChatOpenAI so it should not force the openai
api key to be secret str until openai model has the same changes

https://github.com/langchain-ai/langchain/issues/12841

rlancemartin committed:
Add QA

VinzenzKlass committed:
feat: acquire advisory lock before creating extension in pgvector (#12935)

- **Description:** Acquire advisory lock before attempting to create
extension on postgres server, preventing errors in concurrent
executions.
  - **Issue:** #12933
  - **Dependencies:** None

---------

Co-authored-by: Eugene Yurtsev <eyurtsev@gmail.com>

praveenv committed:
Add SecretStr for Symbl.ai Nebula API (#12896)

Description: This PR masks API key secrets for the Nebula model from
Symbl.ai
Issue: #12165 
Maintainer: @eyurtsev

---------

Co-authored-by: Praveen Venkateswaran <praveen.venkateswaran@ibm.com>

rlancemartin committed:
Add base 64

kacperlukawski committed:
Add support of Cohere Embed v3 (#12940)

Cohere released the new embedding API (Embed v3:
https://txt.cohere.com/introducing-embed-v3/) that treats document and
query embeddings differently. This PR updated the `CohereEmbeddings` to
use them appropriately. It also works with the old models.

rlancemartin committed:
Fmt

rlancemartin committed:
fmt

baskaryan committed:
ChatOpenAI and AzureChatOpenAI openai>=1 compatible (#12948)

Commit:
installation instructions

Commit:
more fixes

rlancemartin committed:
Update invoke and versions

rlancemartin committed:
Merge branch 'master' into rlm/mm_qa_cookbooks

rlancemartin committed:
Merge branch 'master' into rlm/mm_qa_cookbooks

Commit:
new client

Commit:
lineup

rlancemartin committed:
Add GPT-4V

Commit:
x

rlancemartin committed:
Add TODO

Commit:
x

Commit:
x

Commit:
Merge branch 'erick/fix-embeddings' into rlm/mm_qa_cookbooks

Commit:
working!

rlancemartin committed:
Fmt

Commit:
no more dev links

hinthornw commented on cookbook/Multi_modal_RAG.ipynb:
Would be great if you could upload the images somewhere shareable so we don't have to look at e.g., "/Users/erickfriis/Desktop/wildfire_stats/" o

rlancemartin replied:
Yes. The template will address this b/c we can simply package the pdf w/ the template (and store any intermediate files). It's a bit non-ideal w/ cookbooks, which is why we are generally pushing towards templates. Multi-modal template will likely be out ~Web (w/ Chroma and the option to use open_clip embd). But this was a special case where it made sense to get the cookbooks out: https://github.com/langchain-ai/langchain/pull/12859

Commit:
oai v1 embeddings (#12969)

Initial PR to get OpenAIEmbeddings working with the new sdk

fyi @rlancemartin

Fixes #12943

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

ofermend committed:
Vectara RAG template (#12975)

- **Description:** RAG template using Vectara
  - **Twitter handle:** @ofermend

hwchase17 committed:
add ingest for mongo (#12897)

rlancemartin committed:
Merge branch 'master' into rlm/mm_qa_cookbooks

rlancemartin committed:
Final fmt

rlancemartin committed:
Update version
